sale_commission_oca/models/sale_order.py

```diff
@@ -28,16 +29,16 @@ def _compute_commission_total(self):
     @api.depends("partner_agent_ids", "order_line.agent_ids.agent_id")
     def _compute_agents(self):
         for so in self:
-            so.partner_agent_ids = [
-                (6, 0, so.mapped("order_line.agent_ids.agent_id").ids)
-            ]
+            agents = so.mapped("order_line.agent_ids.agent_id")
+            so.partner_agent_ids = [Command.set(agents.ids)]
 
     @api.model
     def _search_agents(self, operator, value):
         sol_agents = self.env["sale.order.line.agent"].search(
             [("agent_id", operator, value)]
         )
-        return [("id", "in", sol_agents.mapped("object_id.order_id").ids)]
+        agents = sol_agents.mapped("object_id.order_id").ids
```

Comment:
Please name variables correctly, and for the relational fields, you can omit `mapped`:

```suggestion
        agent_ids = sol_agents.object_id.order_id.ids
```